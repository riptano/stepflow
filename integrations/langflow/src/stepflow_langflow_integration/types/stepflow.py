"""Stepflow workflow and step type definitions."""

from dataclasses import dataclass, field
from typing import Dict, Any, List, Optional


@dataclass
class StepflowStep:
    """Represents a Stepflow workflow step."""
    
    id: str
    component: str
    input: Dict[str, Any]
    output_schema: Optional[Dict[str, Any]] = None
    
    def to_dict(self) -> Dict[str, Any]:
        """Convert step to dictionary representation."""
        result = {
            "id": self.id,
            "component": self.component,
            "input": self.input,
        }
        
        if self.output_schema:
            result["output_schema"] = self.output_schema
        
        return result


@dataclass 
class StepflowWorkflow:
    """Represents a Stepflow workflow."""
    
    name: str
    steps: List[StepflowStep]
    description: Optional[str] = None
    version: str = "1.0"
    schema: str = "https://stepflow.org/schemas/v1/flow.json"
    output: Optional[Dict[str, Any]] = None
    
    def to_dict(self) -> Dict[str, Any]:
        """Convert workflow to dictionary representation."""
        result = {
            "schema": self.schema,
            "name": self.name,
            "steps": [step.to_dict() for step in self.steps],
        }
        
        if self.description:
            result["description"] = self.description
        
        if self.version != "1.0":
            result["version"] = self.version
        
        if self.output:
            result["output"] = self.output
        
        return result