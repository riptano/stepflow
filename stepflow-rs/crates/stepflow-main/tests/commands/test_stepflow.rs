use super::stepflow;
use insta_cmd::assert_cmd_snapshot;

#[test]
fn test_help() {
    assert_cmd_snapshot!(stepflow().arg("--help"));
}

#[test]
fn test_version() {
    assert_cmd_snapshot!(stepflow().arg("--version"));
}

#[test]
fn test_no_command() {
    assert_cmd_snapshot!(stepflow());
}

#[test]
fn test_invalid_command() {
    assert_cmd_snapshot!(stepflow().arg("invalid-command"));
}

#[test]
fn test_with_log_level() {
    assert_cmd_snapshot!(stepflow().arg("--log-level=debug").arg("--help"));
}

#[test]
fn test_with_different_log_levels() {
    assert_cmd_snapshot!(
        stepflow()
            .arg("--log-level=trace")
            .arg("--other-log-level=debug")
            .arg("--help")
    );
}
