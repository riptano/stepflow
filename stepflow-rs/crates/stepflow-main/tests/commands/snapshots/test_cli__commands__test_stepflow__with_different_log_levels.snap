---
source: crates/stepflow-main/tests/commands/test_stepflow.rs
info:
  program: stepflow
  args:
    - "--log-file=/dev/null"
    - "--omit-stack-trace"
    - "--log-level=trace"
    - "--other-log-level=debug"
    - "--help"
---
success: true
exit_code: 0
----- stdout -----
Stepflow CLI.

Usage: stepflow [OPTIONS] <COMMAND>

Commands:
  run              Run a workflow directly
  serve            Start a StepFlow service
  submit           Submit a workflow to a StepFlow service
  test             Run tests defined in workflow files or directories
  list-components  List all available components from a stepflow config
  repl             Start an interactive REPL for workflow development and debugging
  validate         Validate workflow files and configuration
  help             Print this message or the help of the given subcommand(s)

Options:
      --log-level <LEVEL>        Set the log level for StepFlow [default: info] [possible values: trace, debug, info, warn, error]
      --other-log-level <LEVEL>  Set the log level for other parts of StepFlow [default: warn] [possible values: trace, debug, info, warn, error]
      --log-file <FILE>          Write logs to a file instead of stderr
      --omit-stack-trace         Omit stack traces (line numbers of errors)
  -h, --help                     Print help
  -V, --version                  Print version

----- stderr -----
